api_test.go

```diff
@@ -564,3 +565,75 @@ func TestGetOAuthClients(t *testing.T) {
 
 	ApiSpecRegister = nil
 }
+
+func TestResetHandler(t *testing.T) {
+	uri := "/tyk/reload/"
+
+	ApiSpecRegister = make(map[string]*APISpec)
+
+	makeSampleAPI(t, apiTestDef)
+
+	recorder := httptest.NewRecorder()
+	params := make(url.Values)
+
+	req, err := http.NewRequest("GET", uri+params.Encode(), nil)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	resetHandler(recorder, req)
+
+	if recorder.Code != 200 {
+		t.Fatal("Hot reload failed, response code was: ", recorder.Code)
+	}
+
+	if len(ApiSpecRegister) == 0 {
+		t.Fatal("Hot reload was triggered but no APIs were found.")
+	}
+}
+
+func TestGroupResetHandler(t *testing.T) {
+	signalChan := make(chan bool)
+	cacheStore := RedisClusterStorageManager{}
+	cacheStore.Connect()
+
+	go func(pubSubChannel string, store RedisClusterStorageManager, signalChan chan bool) {
+		store.StartPubSubHandler(pubSubChannel, func(message redis.Message) {
+			notif := Notification{}
+			if err := json.Unmarshal(message.Data, &notif); err != nil {
+				t.Fatal("Unmarshalling message body failed, malformed: ", err)
+			}
+			if notif.Command == NoticeGroupReload {
+				signalChan <- true
```

Comment:
I'd say otherwise - this wastes everyone's time. The test should fail as soon as possible, instead of hitting the timeout on purpose (which by default is 10m).